Add 3.27 release blog post (#2954)
sourcegraph/about: blogposts/2021/release-3.27.md

    url: https://github.com/sourcegraph/sourcegraph/pull/19527
    category: Batch Changes
  - description: "Creating changesets through Batch Changes as a site-admin without configured Batch Changes credentials has been deprecated. To avoid any interruptions in changeset creation, please configure user or global credentials before Sourcegraph 3.29."
    url: https://github.com/sourcegraph/sourcegraph/pull/20143
    category: Admin
  - description: "New changes of a Perforce depot will now be reflected in `master` branch after the initial clone."
    url: https://github.com/sourcegraph/sourcegraph/pull/19718
    category: Admin
  - description: "Search functionality via built-in `contains` predicate: `repo:contains(...)`, `repo:contains.file(...)`, `repo:contains.content(...)`, repo:contains.commit.after(...)`."
   url: https://github.com/sourcegraph/sourcegraph/issues/18584
   category: Search
  - description: "Database encryption, external service config \u0026 user auth data can now be encrypted in the database using the `encryption.keys` config. See the docs for more info."
   url: https://docs.sourcegraph.com/admin/encryption

>    category:

Author: ```suggestion
   category: Admin
```

  - description: "The new `disableAutoCodeHostSyncs` setting allows site admins to disable any periodic background syncing of configured code host connections. That includes syncing of repository metadata (i.e. not git updates, use `disableAutoGitUpdates` for that), permissions and batch changes changesets, but may include other data we'd sync from the code host API in the future."

>    url:
>    category: Admin

Author: ```suggestion
```

---

Sourcegraph 3.27 is now available! For this release, we focused on making Sourcegraph Batch Changes even better, upgraded our minimum supported version of Postgres from 9.6 to 12, and added a new `contains` functionality to search.

## Batch Changes Updates

[Batch Changes](https://about.sourcegraph.com/batch-changes/) allows you to automate large-scale code changes with a single declarative file. With a simple UI, Batch Changes makes it easy to track and manage all of your changesets from checks to code reviews until each change is merged.
As part of this release you can now define batch change rollout windows allowing you to better control when changes are rolled out, and we added the ability to mark changesets as “archived” when a new batch spec is created.

## Postgres update

We updated the minimum supported version of Postgres from 9.6 to 12. If you are maintaining an external database and your Postgres version is older than Postgres 12, you will need to update your database instance prior to upgrading from Sourcegraph 3.26 to 3.27.
[Read more about why we made this change](https://about.sourcegraph.com/blog/postgres-version-update/).

## New "Contains" functionality 

> You can now search for repos containing multiple patterns using the built-in `contains` predicate: `repo:contains(...)`, `repo:contains.file(...)`, `repo:contains.content(...)`, and `repo:contains.commit.after(...)`.

Reviewer: ```suggestion
Contains is a new query built-in that enables you to conditionally filter repositories based on the repository contents. For example, you can now search across repositories only if those repositories contain a particular file (like a README), or a particular file with some content (like a CHANGELOG file that includes the word fix).
```

    url: https://github.com/sourcegraph/sourcegraph/pull/19951
    category: Batch Changes
  - description: "Bumped the minimum supported version of Postgres from `9.6` to `12`. The upgrade procedure is mostly automated for existing deployments, but may require action if using the single-container deployment or an external database. See the upgrade documentation for your deployment type for detailed instructions."
    url: https://docs.sourcegraph.com/admin/updates
    category:
  - description: "Changesets in batch changes will now be marked as archived instead of being detached when a new batch spec that doesn't include the changesets is applied. Once they're archived users can manually detach them in the UI."
    url: https://github.com/sourcegraph/sourcegraph/pull/19527
    category: Batch Changes
  - description: "The default replica count on `sourcegraph-frontend` and `precise-code-intel-worker` for Kubernetes has changed from `1` -\u003e `2`."
    url:
    category:

>   - description: Changes to code monitor trigger search queries
>     url: https://github.com/sourcegraph/sourcegraph/pull/19680
>     category: Search

Author: ```suggestion
```

>   - description: "Deprecated the GraphQL `icon` field on `GenericSearchResultInterface`. It will be removed in a future release."
>     url: https://github.com/sourcegraph/sourcegraph/pull/20028/files
>     category: Search

Author: ```suggestion
```

  - description: "Creating changesets through Batch Changes as a site-admin without configured Batch Changes credentials has been deprecated. Please configure user or global credentials before Sourcegraph 3.29 to not experience any interruptions in changeset creation."
    url: https://github.com/sourcegraph/sourcegraph/pull/20143
    category: Admin

> # Fixed

Author: ```suggestion
```

>   - description: "A regression caused by search onboarding tour logic to never focus input in the search bar on the homepage. Input now focuses on the homepage if the search tour isn't in effect."
>     url: https://github.com/sourcegraph/sourcegraph/pull/19678
>     category: Search

Author: ```suggestion
```

  - description: "New changes of a Perforce depot will now be reflected in `master` branch after the initial clone."
    url: https://github.com/sourcegraph/sourcegraph/pull/19718
    category: Admin

>   - description: Gitolite and Other type code host connection configuration can be correctly displayed.
>     url: https://github.com/sourcegraph/sourcegraph/pull/19976
>     category:
>   - description: Fixed a regression that caused user and code host limits to be ignored.
>     url: https://github.com/sourcegraph/sourcegraph/pull/20089
>     category:

Author: ```suggestion
```

---
title: "Sourcegraph 3.27 release"
publishDate: 2021-04-20T10:00-07:00
description: "Sourcegraph 3.27 is released."
tags: [blog, release]
slug: "release/3.27"
published: true

Sourcegraph 3.27 is now available! For this release, we focused on making Sourcegraph Batch Changes even better, added precise Code Intelligence for C++ and Java, and upgraded our minimum supported version of Postgres from 9.6 to 12.

**Batch Changes Updates** 
[Batch Changes](https://about.sourcegraph.com/batch-changes/) allows you to automate large-scale code changes with a single declarative file. With a simple UI, Batch Changes makes it easy to track and manage all of your changesets from checks to code reviews until each change is merged. 

> As part of this release you can now define Batch Changes rollout windows allowing you to better control when changes are rolled out, and we added the ability to mark changesets as “archived” when a new batch spec is created. 

Reviewer: @content-hacker I think batch changes rollout windows should be lower-cased since we're referring to the functionality vs the specific feature.

**Postgres update**
We updated the minimum supported version of Postgres from 9.6 to 12. If you are maintaining an external database and your Postgres version is older than Postgres 12, you will need to update your database instance prior to upgrading from Sourcegraph 3.26 to 3.27. 
[Read more about why we made this change](https://about.sourcegraph.com/blog/postgres-version-update/). 

> changelogItems:

Reviewer: ```suggestion
```

Author: I think we need this for the proper formatting. Is there a reason you've removed it?

Reviewer: @christinaforney - my mistake - I didn't realize we needed it.

>   # Added

Reviewer: ```suggestion
```

>   - description: "`count:` now supports \"all\" as value. Queries with `count:all` will return up to 999999 results."

Reviewer: @christinaforney @content-hacker - I recommend cutting this update.

Author: agree - I'll make suggested changes and give you a 👍  where I agree.

>   - description: "`count:` now supports \"all\" as value. Queries with `count:all` will return up to 999999 results."
>     url: https://github.com/sourcegraph/sourcegraph/pull/19756
>     category:

Author: ```suggestion
```

>   - description: Credentials for Batch Changes are now validated when adding them.

Reviewer: @christinaforney @content-hacker - I recommend cutting this update.

>   - description: Credentials for Batch Changes are now validated when adding them.
>     url: https://github.com/sourcegraph/sourcegraph/pull/19602
>     category: Batch Changes

Author: ```suggestion
```

  - description: "Batch Changes now ignore repositories that contain a `.batchignore` file. [#19877](https://github.com/sourcegraph/sourcegraph/pull/19877) and src-cli#509"
    url: "https://github.com/sourcegraph/src-cli/pull/509"
    category: Batch Changes

>   - description: "Side-by-side diff for commit visualization."

Reviewer: ```suggestion
  - description: "The diff view commit visualization now supports both split and unified views, including hovers, go-to-definition, and find-references."
```

Reviewer 2: @abschu Beyang tweeted about this with a GIF: https://twitter.com/beyang/status/1384541090636701698 

Should we pull this into the intro/highlights section?

Reviewer: @content-hacker - great idea!

Author: This feels more important to me than the update to batch changes.

    url: https://github.com/sourcegraph/sourcegraph/pull/19553

>     category:

Reviewer: ```suggestion
    category: Search
```

>   - description: "The site configuration now supports defining batch change rollout windows, which can be used to slow or disable pushing changesets at particular times of day or days of the week. [#19796](https://github.com/sourcegraph/sourcegraph/pull/19796), [#19797](https://github.com/sourcegraph/sourcegraph/pull/19797), and #19951."

Reviewer: ```suggestion
  - description: "The site configuration now supports defining batch change rollout windows, which can be used to slow or disable pushing changesets at particular times of day or days of the week."
```

@content-hacker - should we include this in the body (in addition to the intro)?

Reviewer 2: @abschu I think so. I see the intro as a highlights section and the changelog as the "full details" section.

    url: https://github.com/sourcegraph/sourcegraph/pull/19951
    category: Batch Changes

> # Changed

Reviewer: ```suggestion
```

  - description: "Bumped the minimum supported version of Postgres from `9.6` to `12`. The upgrade procedure is mostly automated for existing deployments, but may require action if using the single-container deployment or an external database. See the upgrade documentation for your deployment type for detailed instructions."
    url: https://docs.sourcegraph.com/admin/updates

>     category:

Reviewer: ```suggestion
    category: Admin
```

  - description: "Changesets in batch changes will now be marked as archived instead of being detached when a new batch spec that doesn't include the changesets is applied. Once they're archived users can manually detach them in the UI."
    url: https://github.com/sourcegraph/sourcegraph/pull/19527
    category: Batch Changes

>   - description: "The default replica count on `sourcegraph-frontend` and `precise-code-intel-worker` for Kubernetes has changed from `1` -\u003e `2`."

Reviewer: @content-hacker @christinaforney - I recommend cutting this update.

    url:
    category:

>   - description: Changes to code monitor trigger search queries

Reviewer: @christinaforney @content-hacker - I recommend cutting this.

    url: https://github.com/sourcegraph/sourcegraph/pull/19680
    category: Search

>   - description: "Deprecated the GraphQL `icon` field on `GenericSearchResultInterface`. It will be removed in a future release."

Reviewer: @content-hacker @christinaforney - I recommend cutting this.

    url: https://github.com/sourcegraph/sourcegraph/pull/20028/files
    category: Search

>   - description: "Creating changesets through Batch Changes as a site-admin without configured Batch Changes credentials has been deprecated. Please configure user or global credentials before Sourcegraph 3.29 to not experience any interruptions in changeset creation."

Reviewer: ```suggestion
  - description: "Creating changesets through Batch Changes as a site-admin without configured Batch Changes credentials has been deprecated. To avoid any interruptions in changeset creation, please configure user or global credentials before Sourcegraph 3.29."
```

    url: https://github.com/sourcegraph/sourcegraph/pull/20143
    category: Admin
# Fixed

>   - description: "A regression caused by search onboarding tour logic to never focus input in the search bar on the homepage. Input now focuses on the homepage if the search tour isn't in effect."

Reviewer: @content-hacker @christinaforney - I recommend cutting this update.

    url: https://github.com/sourcegraph/sourcegraph/pull/19678
    category: Search
  - description: "New changes of a Perforce depot will now be reflected in `master` branch after the initial clone."
    url: https://github.com/sourcegraph/sourcegraph/pull/19718

>     category:

Reviewer: @christinaforney - what category is this?

>   - description: Gitolite and Other type code host connection configuration can be correctly displayed.

Reviewer: @content-hacker @christinaforney - I recommend cutting this update.

    url: https://github.com/sourcegraph/sourcegraph/pull/19976
    category:

>   - description: Fixed a regression that caused user and code host limits to be ignored.

Reviewer: @content-hacker @christinaforney - I recommend cutting this update.

  - description: "New changes of a Perforce depot will now be reflected in `master` branch after the initial clone."
    url: https://github.com/sourcegraph/sourcegraph/pull/19718
    category: Admin

>   - description: "Search functionality via built-in `contains` predicate: `repo:contains(...)`, `repo:contains.file(...)`, `repo:contains.content(...)`, repo:contains.commit.after(...)`."

Author: ```suggestion
  - description: "You can now search for repos containing multiple patterns using the built-in `contains` predicate: `repo:contains(...)`, `repo:contains.file(...)`, `repo:contains.content(...)`, and `repo:contains.commit.after(...)`."
```

   url: https://github.com/sourcegraph/sourcegraph/issues/18584
   category: Search

>   - description: "Database encryption, external service config \u0026 user auth data can now be encrypted in the database using the `encryption.keys` config. See the docs for more info."

Reviewer: @christinaforney - recommendation: User auth data can now be encrypted in the database using the `encryption.keys` config. 

Category: Admin

Author: ```suggestion
  - description: "User auth data can now be encrypted in the database using the `encryption.keys` config."
```

   url: https://docs.sourcegraph.com/admin/encryption
   category:

>   - description: "The new `disableAutoCodeHostSyncs` setting allows site admins to disable any periodic background syncing of configured code host connections. That includes syncing of repository metadata (i.e. not git updates, use `disableAutoGitUpdates` for that), permissions and batch changes changesets, but may include other data we'd sync from the code host API in the future."

Reviewer: @christinaforney :  "The new `disableAutoCodeHostSyncs` setting allows site admins to disable any periodic background syncing of configured code host connections. This includes syncing of repository metadata (use `disableAutoGitUpdates` to disable git updates, ), permissions, and batch changes changesets, but may include other data we sync from the code host API in the future."

Author: I actually think we shouldn't include this one.

```suggestion
```
